lib/astutils.cpp

```diff
@@ -1364,6 +1364,8 @@ bool isVariableChangedByFunctionCall(const Token *tok, int indirect, const Setti
     tok = getTokenArgumentFunction(tok, argnr);
     if (!tok)
         return false; // not a function => variable not changed
+    if (tok->isKeyword() && tok->str().find("_cast") == std::string::npos)
```

Comment:
I would prefer that for instance `isCPPCast` is used instead. If we add some configuration of casts in the future so we can understand `qobject_cast` and other casts better.. then it would be good that this code handle that also.

Reply:
The problem is `isCPPCast` wont return true on the keywords, just on the `(`. I could add a function `isCPPCastKeyword` and then this function and `isCPPCast` can use it.

Reply:
I suggest that `tok->str().find("_cast") == std::string::npos` is replaced with `endsWith(tok->str(), "_cast", 5)`